hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusCreateTopicHandler.java

```diff
@@ -73,20 +79,65 @@ public void preHandle(@NonNull final PreHandleContext context) {
      *
-     * TODO: Provide access to writable topic store.
      *
-     * @param handleContext the {@link HandleContext} for the active transaction
-     * @param topicCreation the {@link ConsensusCreateTopicTransactionBody} for the active transaction
+     * @param handleContext          the {@link HandleContext} for the active transaction
+     * @param op                     the {@link ConsensusCreateTopicTransactionBody} of the active transaction
      * @param consensusServiceConfig the {@link ConsensusServiceConfig} for the active transaction
-     * @param recordBuilder the {@link ConsensusCreateTopicRecordBuilder} for the active transaction
+     * @param recordBuilder          the {@link ConsensusCreateTopicRecordBuilder} for the active transaction
      * @throws NullPointerException if one of the arguments is {@code null}
      */
     public void handle(
             @NonNull final HandleContext handleContext,
-            @NonNull final ConsensusCreateTopicTransactionBody topicCreation,
+            @NonNull final ConsensusCreateTopicTransactionBody op,
             @NonNull final ConsensusServiceConfig consensusServiceConfig,
-            @NonNull final ConsensusCreateTopicRecordBuilder recordBuilder) {
-        throw new UnsupportedOperationException("Not implemented");
+            @NonNull final ConsensusCreateTopicRecordBuilder recordBuilder,
+            @NonNull final WritableTopicStore topicStore) {
+        final var builder = new TopicBuilderImpl();
+
+        /* validate admin and submit keys and set them */
+        if (op.hasAdminKey()) {
+            handleContext.attributeValidator().validateKey(op.getAdminKey());
+            builder.adminKey(asHederaKey(op.getAdminKey()).get());
+        }
+        if (op.hasSubmitKey()) {
+            handleContext.attributeValidator().validateKey(op.getSubmitKey());
+            builder.submitKey(asHederaKey(op.getSubmitKey()).get());
+        }
+
+        /* validate if the current topic can be created */
+        if (topicStore.sizeOfState() >= consensusServiceConfig.maxTopics()) {
+            throw new HandleStatusException(MAX_ENTITIES_IN_PRICE_REGIME_HAVE_BEEN_CREATED);
+        }
+
+        /* validate the topic memo */
+        handleContext.attributeValidator().validateMemo(op.getMemo());
+        builder.memo(op.getMemo());
+
+        /* validate the auto-renew account */
+        final var autoRenewPeriod =
+                op.hasAutoRenewPeriod() ? op.getAutoRenewPeriod().getSeconds() : NA;
+        final var autoRenewAccountId =
+                op.hasAutoRenewAccount() ? op.getAutoRenewAccount().getAccountNum() : NA;
+
+        final var entityExpiryMeta = new ExpiryMeta(
+                NA, // expiry not set explicitly
+                autoRenewPeriod,
+                autoRenewAccountId);
+
+        final var effectiveExpiryMeta = handleContext.expiryValidator().validateCreationAttempt(true, entityExpiryMeta);
```

Comment:
Topics can't "self-fund" like accounts or contracts can, so we should pass `false` here.

Reply:
Agreed. Thank you ! 🙌